messaging/src/main/java/org/axonframework/eventhandling/processors/streaming/pooled/Coordinator.java

```diff
@@ -938,8 +943,8 @@ private Map<Segment, TrackingToken> claimNewSegments() {
                     try {
                         TrackingToken token = joinAndUnwrap(
                                 unitOfWorkFactory.create()
-                                                 .executeWithResult(context -> CompletableFuture.completedFuture(
-                                                         tokenStore.fetchToken(name, segment)))
+                                                 .executeWithResult(context ->
+                                                         tokenStore.fetchToken(name, segment, null))
```

Comment:
```suggestion
                                                         tokenStore.fetchToken(name, segment, context))
```

We should pass the `ProcessingContext` here.